### PR TITLE
Switch client and test usage to apps/v1

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Switches test manifest and client usage from extensions/v1beta1 to apps/v1

* Switches all calls to `.ExtensionsV1beta1().` to `.AppsV1().` (except for ingresses and specific tests around extensions/v1beta1 behavior like deployment rollback and selector mutability)
* Switches all test manifests from extensions/v1beta1 and apps/v1beta* to apps/v1 (except for ingresses and specific test fixtures)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref https://github.com/kubernetes/kubernetes/issues/43214
includes https://github.com/kubernetes/kubernetes/pull/66716

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by justinsb** [comment]
Looks good, but question on the e2e... does this leave us with any coverage on the `extensions/v1beta1` endpoints?

**Comment by liggitt** [comment]
> question on the e2e... does this leave us with any coverage on the `extensions/v1beta1` endpoints?

this preserves tests that exercise deployment rollback (specific to extensions/v1beta1 deployments) and selector mutability (specific to extensions/v1beta1 deployments and replicasets)

**Comment by liggitt** [comment]
@kow3ns any suggestions on a reviewer?

**Comment by liggitt** [comment]
rebased

**Comment by liggitt** [comment]
[TestVolumeProvision flake](https://github.com/kubernetes/kubernetes/issues/70180)
/retest

**Comment by AishSundar** [comment]
@liggitt how critical is this for 1.13? we are in code freeze and if possible, I would like to defer this until 1.14. let me know otherwise.

**Comment by liggitt** [comment]
@kow3ns for apps manifests, @soltysh for kubectl updates

**Comment by liggitt** [comment]
split add-on manifest changes out to https://github.com/kubernetes/kubernetes/pull/72203 for easier review

/uncc @kow3ns

File: pkg/controller/disruption/disruption_test.go [hunk 1]
```diff
@@ -23,9 +23,8 @@ import (
 	"testing"
 	"time"
 
-	apps "k8s.io/api/apps/v1beta1"
+	apps "k8s.io/api/apps/v1"
```
Comment (soltysh) [inline pkg/controller/disruption/disruption_test.go]: Honestly, I'd prefer being explicit in the code with these imports, iow. `appsv1` should be used here. This way it's explicit what you're doing in the code.

File: test/e2e/storage/persistent_volumes-local.go [hunk 2]
```diff
@@ -1627,15 +1626,15 @@ func createProvisionerDaemonset(config *localTestConfig) {
 	provisionerPrivileged := true
 	mountProp := v1.MountPropagationHostToContainer
 
-	provisioner := &extv1beta1.DaemonSet{
+	provisioner := &appsv1.DaemonSet{
```
Comment (soltysh) [inline test/e2e/storage/persistent_volumes-local.go]: This is what I'm talking about, it's reads better :+1:

**Review by soltysh — APPROVED** [review]
/lgtm

File: pkg/controller/disruption/disruption_test.go (hunk 1 above)
Comment (liggitt) [reply pkg/controller/disruption/disruption_test.go]: will make a note for the future, was trying to keep the diff minimal for this PR

**Comment by liggitt** [comment]
the dumpclusterlogs failure in pull-kubernetes-e2e-gke is happening across pulls, got the signal I wanted from all the actual tests in that job passing

**Pull request merged**